File: Keeper/Services/ProfilesService.cs
namespace Keeper.Services
{
  public class ProfilesService
  {
    private readonly ProfilesRepository _repo;

    public ProfilesService(ProfilesRepository repo)
    {
      _repo = repo;
    }

    internal Profile GetProfile(string id)
    {
      Profile profile = _repo.GetProfile(id);
      return profile;
    }

    internal List<Vault> GetUsersVaults(string id)
    {
      List<Vault> vaults = _repo.GetUsersVaults(id);
      return vaults;
    }


    internal List<Keep> GetUsersKeeps(string id)
    {
      List<Keep> keeps = _repo.GetUsersKeeps(id);
      return keeps;
    }
  }
}

File: Keeper.client/src/services/ProfilesService.js
import { AppState } from "../AppState.js"
import { api } from "./AxiosService.js"


class ProfilesService{

  async getProfile(profileId){
    const res = await api.get('api/profiles/' + profileId)
    AppState.profile = res.data
  }

  async getUsersKeeps(profileId){
    const res = await api.get(`api/profiles/${profileId}/keeps`)
    AppState.keeps = res.data
  }

  async getUsersVaults(profileId){
    const res = await api.get(`api/profiles/${profileId}/vaults`)
    AppState.vaults = res.data
  }

}

export const profilesService = new ProfilesService()